lib/experimental/cmdline.nim

```diff
@@ -0,0 +1,2856 @@
+      .initCli()
+    let strFlag = cli.flagBuilder()
+      .name("s")
+      .alias("str", "string")
+      .parser(string, (_, val, var str) => (str = val))
+      .addTo(cli)
+    doAssert cli.longNameOf(strFlag) == some("str")
+
+  result = none string
+  for name in cli.namesOf(flag):
+    if name.len > 1:
+      return some name
+
+func shortNameOf*(cli: Cli, flag: FlagId): Option[string] =
+  ## Returns the first short name of `flag`, if one exists.
+  ##
+  ## A short name is defined as a one-character long name.
+  runnableExamples:
+    import std/options
+    import std/sugar
+
+    var cli = commandBuilder(string)
+      .initCli()
+    let strFlag = cli.flagBuilder()
+      .name("string")
+      .alias("str", "s")
+      .parser(string, (_, val, var str) => (str = val))
+      .addTo(cli)
+    doAssert cli.shortNameOf(strFlag) == some("s")
+
+  result = none string
+  for name in cli.namesOf(flag):
+    if name.len == 1:
+      return some name
+
+func usageOf*(cli: Cli, command: CommandId): lent string =
+  ## Returns the `usage` for `command`, as described with `describe`.
+  runnableExamples:
+    import std/sugar
+
+    let cli = commandBuilder(string)
+      .describe("some usage")
+      .initCli()
+    doAssert cli.usageOf(RootCommand) == "some usage"
+
+  cli.usage[ParameterId command]
+
+func usageOf*(cli: Cli, flag: FlagId): lent string =
+  ## Returns the `usage` for `flag`, as described with `describe`.
+  runnableExamples:
+    import std/sugar
+
+    var cli = commandBuilder(string)
+      .initCli()
+    let strFlag = cli.flagBuilder()
+      .name("string")
+      .describe("a string")
+      .parser((_, v, var s) => (s = v))
+      .addTo(cli)
+    doAssert cli.usageOf(strFlag) == "a string"
+
+  cli.usage[ParameterId flag]
+
+func usageOf*(cli: Cli, positional: PositionalId): lent string =
+  ## Returns the `usage` for `positional`, as described with `describe`.
+  runnableExamples:
+    import std/sugar
+
+    var cli = commandBuilder(string)
+      .initCli()
+    let strPos = cli.positionalBuilder()
+      .name("STR")
+      .describe("a string")
+      .parser((v, var s) => (s = v))
+      .addTo(cli)
+    doAssert cli.usageOf(strPos) == "a string"
+
+  cli.usage[ParameterId positional]
+
+func placeholderOf*(cli: Cli, flag: FlagId): string =
+  ## Returns the `placeholder` for `flag`, as described with `describe`.
+  runnableExamples:
+    import std/sugar
+
+    var cli = commandBuilder(string)
+      .initCli()
+    let strFlag = cli.flagBuilder()
+      .name("string")
+      .describe("a string", "STR")
+      .parser((_, v, var s) => (s = v))
+      .addTo(cli)
+    let str2Flag = cli.flagBuilder()
+      .name("string2")
+      .describe("a string")
+      .parser((_, v, var s) => (s = v))
+      .addTo(cli)
+    doAssert cli.placeholderOf(strFlag) == "STR"
+    doAssert cli.placeholderOf(str2Flag) == ""
+
+  cli.placeholder[ParameterId flag]
+
+func isDispatcher*(cli: Cli, command: CommandId): bool =
```

Comment:
An alias of `hasSubcommands` seems appropriate, the dispatcher terminology is mentioned elsewhere, but I don't think it'd come to mind.

Reply:
I think I might just phase the dispatcher terminology out.